UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.FunctionalTests/FunctionalTests/ExchangeSetByProductVersion.cs

```diff
@@ -0,0 +1,148 @@
+
+            var errorMessage = await apiresponse.ReadAsTypeAsync<ErrorDescriptionResponseModel>();
+            Assert.IsTrue(errorMessage.Errors.Any(e => e.Source == "RequestBody"));
+            Assert.IsTrue(errorMessage.Errors.Any(e => e.Description == "Either body is null or malformed."));
+        }
+
+        [TestCase(null,4,1, "ProductVersions", "Product Versions product name cannot be blank or null.", TestName = "When product name is null.")]
+        [TestCase("", 4, 1, "ProductVersions", "Product Versions product name cannot be blank or null.", TestName = "When product name is blank.")]
+        [TestCase("AU895561", null, 1, "ProductVersions", "Product Versions edition number cannot be less than zero or null.", TestName = "When edition number is null.")]
+        [TestCase("AU895561", 4, null, "ProductVersions", "Product Versions update number cannot be less than zero or null.", TestName = "When update number is null.")]
+        [TestCase("AU895561", -1, 1, "ProductVersions", "Product Versions edition number cannot be less than zero or null.", TestName = "When edition number is less than zero.")]
+        [TestCase("AU895561", 4, -1, "ProductVersions", "Product Versions update number cannot be less than zero or null.", TestName = "When update number is less than zero.")]
+        public async Task WhenICallTheApiWithNullProductVersion_ThenABadRequestStatusIsReturned(string productname, int? editionnumber, int? updatenumber, string sourcemessage, string descriptionmessage)
+        {
+            List<ProductVersionModel> ProductVersiondata = new List<ProductVersionModel>();
+
+            ProductVersiondata.Add(Datahelper.GetProductVersionModelData(productname, editionnumber, updatenumber));
+
+            var apiresponse = await ExchangesetApiClient.GetProductVersionsAsync(ProductVersiondata, "http://fss.ukho.gov.uk/batch/7b4cdf10-adfa-4ed6-b2fe-d1543d8b7272");
+            Assert.AreEqual(400, (int)apiresponse.StatusCode, $"Exchange Set for Product version is  returned {apiresponse.StatusCode}, instead of the expected status 400.");
+
+            var errorMessage = await apiresponse.ReadAsTypeAsync<ErrorDescriptionResponseModel>();
+            Assert.IsTrue(errorMessage.Errors.Any(e => e.Source == sourcemessage));
+            Assert.IsTrue(errorMessage.Errors.Any(e => e.Description == descriptionmessage));
+
+        }
+
+        [TestCase("fss.ukho.gov.uk", TestName = "Callback URL without http or https")]
+        [TestCase("https:/fss.ukho.gov.uk", TestName = "Callback URL with wrong https parameter")]
+        [TestCase("http:/fss.ukho.gov.uk", TestName = "Callback URL with wrong http parameter")]
+        [TestCase("https://", TestName = "Callback URL with only https parameter")]
+        public async Task WhenICallTheApiWithAValidProductVersionWithInvalidCallbackURI_ThenASuccessStatusIsReturned(string callbackurl)
```

Comment:
This is asserting that the response is a 400 not a 2XX response, should this be renamed to ThenAErrorStatusIsReturned?

Reply:
Renamed the test name as "WhenICallTheApiWithAValidProductVersionWithInvalidCallbackURI_ThenABadRequestStatusIsReturned"
Thanks Rob.